create_base_image/create_base_image_hostlib.sh

```diff
@@ -0,0 +1,222 @@
+#!/bin/bash
+
+# Common code to build a host image on GCE
+
+source "../external/shflags/shflags"
+
+# Build instance info
+DEFINE_string build_instance \
+  "${USER}-build" "Instance name to create for the build" "i"
+DEFINE_string build_project "$(gcloud config get-value project)" \
+  "Project to use for scratch"
+DEFINE_string build_zone "$(gcloud config get-value compute/zone)" \
+  "Zone to use for scratch resources"
+DEFINE_string build_tags "" "Tags to add to the GCE instance"
+
+# Build artifacts info
+DEFINE_string build_branch "aosp-master" \
+  "Branch to extract build artifacts" "b"
+DEFINE_string build_target "aosp_cf_x86_phone-userdebug" \
+  "Target device to extract build artifacts"
+DEFINE_string build_id "" \
+  "Build id used to extract build artifacts"
+
+# New image info
+DEFINE_string dest_image "" \
+  "Image to create" "o"
+DEFINE_string dest_family "" \
+  "Image family to add the image to" "f"
+DEFINE_string dest_project "$(gcloud config get-value project)" \
+  "Project to use for the new image" "p"
+
+# Base image info
+DEFINE_string source_image_family "debian-10" \
+  "Image family to use as the base" "s"
+DEFINE_string source_image_project "debian-cloud" \
+  "Project holding the base image" "m"
+
+# CF repo info
+DEFINE_string repository_url \
+  "https://github.com/google/android-cuttlefish.git" \
+  "URL to the repository with host changes" "u"
+DEFINE_string repository_branch master \
+  "Branch to check out"
+
+
+
+fatal_echo() {
+  echo "$1"
+  exit 1
+}
+
+# Sleeps while trying to connect to instance
+wait_for_instance() {
+  alive=""
+  while [[ -z "${alive}" ]]; do
+    sleep 5
+    alive="$(gcloud compute ssh "$@" -- uptime || true)"
+  done
+}
+
+# Gets debian packages from Cuttlefish repo
+fetch_cf_package() {
+
+  local url="$1"
+  local branch="$2"
+  local repository_dir="${url/*\//}"
+  local debian_dir="$(basename "${repository_dir}" .git)"
+
+  git clone "${url}" -b "${branch}"
+  dpkg-source -b "${debian_dir}"
+  rm -rf "${debian_dir}"
+
+  # Gets cf-common version
+  for dsc in *dsc; do
+    CF_VER=$(basename "${dsc/*_/}" .dsc)
+    CF_VER="${CF_VER//\./-}"
+  done
```

Comment:
I don't understand what you're trying to do here.

On every iteration of the loop the same (global?) variable gets overwritten. Why do it in a loop if only the last iteration will be meaningful?

Reply:
I'll change it to an array instead of a for loop, there is just one `.dsc` file 👍